cli/cmd/inject.go

```diff
@@ -31,14 +31,16 @@ const (
 )
 
 type injectOptions struct {
-	disableIdentity bool
+	disableIdentity    bool
+	enableDebugSidecar bool
```

Comment:
what about adding `enableDebugSidecar ` to `proxyConfigOptions`? i think this would enable supporting `linkerd install --enable-debug-sidecar`, which could be useful for debugging the control-plane?

Reply:
I think the goal in https://github.com/linkerd/linkerd2/issues/2199#issuecomment-481775293 was to have this functionality only for when we inject pods. So everything but the control plane. I could definitely see a use for having the control plane having this container.

Another reason why I didn't add this `proxyConfigOptions` is that it the value gets saved in the linkerd config map. So, if I ran this initially with the control plane, any other injected container that gets added after would also get the debug container, which may not be what we want.

Reply:
Ah, yeah agree this should not end up in `linkerd-config`. It looks like based on your recent handling via inject, you could enable similar functionality in `linkerd install` and still avoid the `linkerd-config` issue. This is all fine to defer or do in a subsequent PR.